### ISSUE TITLE
json_array_contains() get wrong results when the values in json array cast fail

### ISSUE THREAD

**Issue by xiaodouchen** [facebookincubator/velox#6438]
### Bug description

Test cases:
```
// Presto TestJsonFunctions.java
assertFunction("JSON_ARRAY_CONTAINS('[92233720368547758071]', -9)", BOOLEAN, false);
assertFunction("JSON_ARRAY_CONTAINS('[9.6E400]', 4.2)", BOOLEAN, false);

// Velox 
EXPECT_EQ(jsonArrayContains<int64_t>(R"([92233720368547758071])", -9), std::nullopt);
EXPECT_EQ(jsonArrayContains<double>(R"([9.6E400])", 4.2), std::nullopt);
```

There's a bug in the implementation of `SIMDJsonArrayContainsFunction`. In the code, `v.get_bool(), v.get_int64(), v.get_uint64(), v.get_double(), v.get_string()` may report an error due to the failure of casting the json element in json array to the corresponding type of the containing value. The corresponding error type is simdjson:: INCORRECT_TYPE and simdjson::NUMBER_ERROR. However, this is not a failure to parse json, and null should not be returned directly.

### System information

Velox System Info v0.0.2
Commit: aa1728f95c84307679d9d9d74aa583a96b81d189
CMake Version: 3.25.1
System: Darwin-21.4.0
Arch: arm64
C++ Compiler: /Library/Developer/CommandLineTools/usr/bin/c++
C++ Compiler Version: 13.1.6.13160021
C Compiler: /Library/Developer/CommandLineTools/usr/bin/cc
C Compiler Version: 13.1.6.13160021
CMake Prefix Path: /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr;/opt/homebrew;/usr/local;/usr;/;/opt/homebrew/Cellar/cmake/3.25.1;/usr/local;/usr/X11R6;/usr/pkg;/opt;/sw;/opt/local

**Comment by mbasmanova** [comment]
CC: @kevinwilfong @kagamiori @laithsakka

**Comment by mbasmanova** [comment]
@xiaodouchen Would you help fix this and #6437?

**Comment by xiaodouchen** [comment]
@mbasmanova Yes, I am working on it.
